addons/pos_hr/models/hr_employee_public.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class HrEmployeePublic(models.Model):
+    _inherit = "hr.employee.public"
+
+    def read(self, fields):
+        # as `pos_blackbox_be` is a certified module, it's hard to make fixes in it
+        # so this is a workaround to remove `insz_or_bis_number` field from the fields list
+        # as the parent hr.employee model will attempt to read it from hr.employee.public
+        # where it doesn't exist
+        if 'insz_or_bis_number' in fields:
+            pos_blackbox_be_installed = self.env['ir.module.module'].sudo().search_count([('name', '=', 'pos_blackbox_be'), ('state', '=', 'installed')])
+            had_hr_user_group = self.env.user.has_group('hr.group_hr_user')
+            if not pos_blackbox_be_installed or not had_hr_user_group:
```

Comment:
```suggestion
            if pos_blackbox_be_installed and not had_hr_user_group:
```

I believe the logic should be: "if blackbox_is_installed and has_no_hr_group, then remove the field". If blackbox is not installed, we can't just remove the field from the list, it might be coming from a third party module.

Reply:
@caburj  good catch ! thank you :pray: